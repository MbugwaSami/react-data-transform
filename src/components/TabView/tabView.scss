.rdt-tabs {
    height: 100%;
    display: flex;
    flex-direction: column;

    .rdt-tabs-header {
        display: flex;
        flex-wrap: wrap;
        background-color: #0858a8;

        .rdt-tab-item {
            display: flex;
            align-items: center;
            justify-content: flex-start;
            cursor: pointer;
            min-width: 100px;
            padding: 0 5px;
            border-color: #356faa;
            border-style: solid;
            border-width: 0 1px;
            color: #ffffff;

            &:first-child {
                border-left: none;
            }

            &:last-child {
                border-right: none;
            }

            .rdt-tab-item-icon {
                align-self: end;

                svg {
                    fill: #ffffff;
                }
            }


        }

        .rdt-tab-item-selected {
            background-color: #e9e9e9;
            color: #0858a8;

            .rdt-tab-item-icon {
                svg {
                    fill: #0858a8;
                }
            }

        }

    }

    .rdt-tabs-content {
        flex-grow: 1
    }
}