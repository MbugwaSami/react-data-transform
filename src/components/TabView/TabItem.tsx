import { FC } from "react";

import "./tabView.scss";

const TabItem: FC<TabItemProps> = (props) => {
  const { title, onClick, active, styles, icon } = props;

  return (
    <div
      style={styles}
      aria-label={title}
      tabIndex={0}
      role="presentation"
      className={active ? "rdt-tab-item-selected rdt-tab-item" : "rdt-tab-item"}
      onClick={onClick}
      onKeyPress={(e) => {
        if (e.key === "Enter" || e.key === " ") {
          onClick();
        }
      }}
    >
      <div className="rdt-tab-item-icon">{icon && icon}</div>
      <h4>{title}</h4>
    </div>
  );
};

export default TabItem;
