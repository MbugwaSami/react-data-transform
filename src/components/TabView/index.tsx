import { FC, useState, useRef } from "react";

import "./tabView.scss";
import TabItem from "./TabItem";

const TabView: FC<TabsViewProps> = (props) => {
  const { tabs, children, styles } = props;
  const [currentTab, setCurrentTab] = useState<number>(0);
  const bodyRef = useRef<HTMLDivElement>(null);

  const handleClick = (current: number) => {
    bodyRef?.current?.focus();
    setCurrentTab(current);
  };

  return (
    <div style={styles?.tabView} className="rdt-tabs">
      <div style={styles?.tabViewHeader} className="rdt-tabs-header">
        {tabs?.map((tab, index) => (
          <TabItem
            key={`Tab ${index + 1}`}
            onClick={() => handleClick(index)}
            {...tab}
            active={index === currentTab}
            styles={styles?.tabItem}
          />
        ))}
      </div>
      <div
        role="presentation"
        ref={bodyRef}
        tabIndex={0}
        className="rdt-tabs-content"
      >
        {children[currentTab]}
      </div>
    </div>
  );
};

export default TabView;
