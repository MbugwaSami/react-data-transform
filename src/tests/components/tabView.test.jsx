import React from "react";
import { shallow, mount } from "enzyme";
import TabView from "../../components/TabView";
import { tabViewData } from "../mockData";

describe("<TabView Tests />", () => {
  const setState = jest.fn();
  const useStateMock = (initState) => [initState, setState];
  afterEach(() => {
    jest.clearAllMocks();
  });
  it("should render <TabView />", () => {
    const wrapper = shallow(<TabView {...tabViewData} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("should render all tab items", () => {
    const wrapper = mount(<TabView {...tabViewData} />);
    const tabElement = wrapper.find(".rdt-tab-item");
    expect(tabElement).toHaveLength(tabViewData.tabs.length);
  });
  it("should change tabs", () => {
    jest.spyOn(React, "useState").mockImplementation(useStateMock);
    const wrapper = mount(<TabView {...tabViewData} />);
    const tabElement = wrapper.find(".rdt-tab-item").at(1);
    tabElement.simulate("click");
    expect(setState).toHaveBeenCalledWith(1);
  });
});
