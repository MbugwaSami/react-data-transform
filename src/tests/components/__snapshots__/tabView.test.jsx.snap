// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TabView Tests /> should render <TabView /> 1`] = `
<div
  className="rdt-tabs"
>
  <div
    className="rdt-tabs-header"
  >
    <TabItem
      active={true}
      key="Tab 1"
      onClick={[Function]}
      title="test"
    />
    <TabItem
      active={false}
      key="Tab 2"
      onClick={[Function]}
      title="test 2"
    />
  </div>
  <div
    className="rdt-tabs-content"
    role="presentation"
    tabIndex={0}
  />
</div>
`;
